### PR TITLE
Added new relic browser

### PR DESCRIPTION
**Pull request by theramis**
Just wanted to give this a go and see what we get. 

No other reason than curiosity tbh. 
Happy to not have it if its not a good idea 😅

### CODE REVIEW

File: app/newRelicSnippet.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+;NREUM.loader_config={accountID:"3340026",trustKey:"3340026",agentID:"1385887337",licenseKey:"NRJS-94c48cdf4bf4fa5718d",applicationID:"1385887337"}
+;NREUM.info={beacon:"bam.nr-data.net",errorBeacon:"bam.nr-data.net",licenseKey:"NRJS-94c48cdf4bf4fa5718d",applicationID:"1385887337",sa:1}
+`;
+
+export const NewRelicSnippet = () => {
+  return (
+    <script
```
Comment (andycarrell) [inline app/newRelicSnippet.tsx:11]: Thoughts on using next/Script here? https://nextjs.org/docs/basic-features/script

**Comment by andycarrell** [comment]
What's it give us? Analytics and error reporting?

**Comment by theramis** [comment]
> What's it give us? Analytics and error reporting?

Quite a few things. 
errors, performance, everything GA gives us  

image

**Comment by jishaal** [comment]
Its cool! Not sure if we need it? Don't mind either way haha if we just want to use it as a means to explore

File: app/newRelicSnippet.tsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+;NREUM.loader_config={accountID:"3340026",trustKey:"3340026",agentID:"1385887337",licenseKey:"NRJS-94c48cdf4bf4fa5718d",applicationID:"1385887337"}
+;NREUM.info={beacon:"bam.nr-data.net",errorBeacon:"bam.nr-data.net",licenseKey:"NRJS-94c48cdf4bf4fa5718d",applicationID:"1385887337",sa:1}
```
Comment (jishaal) [inline app/newRelicSnippet.tsx:8]: Would we want keys between local/deploy-preview and prod to be different?

Comment (theramis) [reply app/newRelicSnippet.tsx:8]: yea I wasn't sure about that. 
tbh I don't think it matters too much

**Comment by theramis** [comment]
> Its cool! Not sure if we need it? Don't mind either way haha if we just want to use it as a means to explore

tbh I don't reallllly need it. 
Was just looking at new relic pricing and found out it was free so thought why not haha

File: app/newRelicSnippet.tsx (hunk 2 above)
Comment (andycarrell) [reply app/newRelicSnippet.tsx:6]: IMO we should. Can be as simple as not rendering the script locally?

File: app/newRelicSnippet.tsx (hunk 1 above)
Comment (theramis) [reply app/newRelicSnippet.tsx:11]: So I tried by just converting this to Script. (I might be doing it wrong)
I can't seem to get it to load on the page though. 

I tried even just adding this 

image


Added that to the _document, _app, index.tsx. 

Nothing. What am I doing wrong?

File: app/newRelicSnippet.tsx (hunk 2 above)
Comment (theramis) [reply app/newRelicSnippet.tsx:6]: Whats the best way to do that?

Is there an inbuilt way or should I use env variables?

Comment (andycarrell) [reply app/newRelicSnippet.tsx:6]: @theramis just use Env variables. Probably can check if it's production at this point

File: app/newRelicSnippet.tsx (hunk 1 above)
Comment (theramis) [reply app/newRelicSnippet.tsx]: nvm figured it out, I was testing it incorrectly 😓

File: app/newRelicSnippet.tsx (hunk 2 above)
Comment (theramis) [reply app/newRelicSnippet.tsx:8]: done

File: app/newRelicSnippet.tsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  if (process.env.NEXT_PUBLIC_ENVIRONMENT === "production") {
+    return (
+      <Script
+        id="new-relic-snippet"
+        dangerouslySetInnerHTML={{ __html: snippet }}
+      />
+    );
+  }
+  return null;
+};
```
Comment (andycarrell) [inline app/newRelicSnippet.tsx:21]: Damn this file name should be `NewRelicSnippet`

**Review by andycarrell — APPROVED** [review]

**Pull request merged**